### PR TITLE
Query: Fixes substatus code for GROUP BY queries in PartitionRoutingHelper

### PR DESCRIPTION
**Pull request by neildsh**
## Description

Currently PartitionRoutingHelper.GetProvidedPartitionKeyRanges does not correctly set the substatus code for group by queries. This change also exposes allowNonValueAggregates as a parameter in this method, whereas it was previously hardcoded to false. This meant that Non value aggregate queries would not get back a proper substatus code either. This new parameter will be used by the Compute Gateway.

Other SDKs are supposed to use this substatus code to fetch a query plan. In the absence of this, certain hacks have been put into the NodeJS SDk for example where they check if the error text contains certain strings. as we move from Routing gateway to compute Gateway, some of these strings may change since the v3 sdk has slightly different code paths from the v2 sdk and it also localizes some error messages.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/Routing/PartitionRoutingHelperTest.cs [hunk 1]
```diff
@@ -16,18 +16,43 @@ namespace Microsoft.Azure.Cosmos.Tests.Routing
...
 
     /// <summary>
     /// Tests for <see cref="PartitionRoutingHelper"/> class.
     /// </summary>
     [TestClass]
     public class PartitionRoutingHelperTest
     {
-        PartitionRoutingHelper partitionRoutingHelper;
-        public PartitionRoutingHelperTest()
+        private static readonly Dictionary<string, object> DefaultQueryEngineConfiguration = new Dictionary<string, object>()
```
Comment (sboshra) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/Routing/PartitionRoutingHelperTest.cs:29]: >DefaultQueryEngineConfiguration [](http://example.com/codeflow?start=59&length=31)

make this shared across all tests

Comment (sboshra) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/Routing/PartitionRoutingHelperTest.cs:29]: >Dictionary [](http://example.com/codeflow?start=32&length=10)

ReadOnlyDictionary

**Review by sboshra — DISMISSED** [review]
:shipit:

File: Microsoft.Azure.Cosmos/src/Routing/PartitionRoutingHelper.cs [hunk 2]
```diff
@@ -36,6 +36,33 @@ internal class PartitionRoutingHelper
             QueryPartitionProvider queryPartitionProvider,
             string clientApiVersion,
             out QueryInfo queryInfo)
+        {
+            return PartitionRoutingHelper.GetProvidedPartitionKeyRanges(
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/Routing/PartitionRoutingHelper.cs:40]: Please add a clear description of what scenarios can hit this issue and what the side effects are of it. Does the customer see 500s, invalid query results, etc...?

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/Routing/PartitionRoutingHelperTest.cs (hunk 1 above)
Comment (neildsh) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/Routing/PartitionRoutingHelperTest.cs]: done

Comment (neildsh) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/Routing/PartitionRoutingHelperTest.cs]: changed

File: Microsoft.Azure.Cosmos/src/Routing/PartitionRoutingHelper.cs (hunk 2 above)
Comment (neildsh) [reply Microsoft.Azure.Cosmos/src/Routing/PartitionRoutingHelper.cs]: updated the PR description

**Review by sboshra — APPROVED** [review]

**Pull request merged**